### PR TITLE
chore: add release workflow

### PR DESCRIPTION
**Pull request by Ian2012**
This PR adds a release workflow to upload the tutor-contrib-harmony plugin to pypi.

This was tested on a fork: https://github.com/eduNEXT/openedx-k8s-harmony/actions/runs/12011623600/job/33481030042, it failed because the PyPi token was missing

### CODE REVIEW

File: .github/workflows/pypi-publish.yaml
Comment (Ian2012) [inline .github/workflows/pypi-publish.yaml:11]: Only runs for vX.X.X tags (ignore harmony-chart-X.X.X)

Comment (bradenmacdonald) [reply .github/workflows/pypi-publish.yaml:11]: Might be easier to just put it in a separate repo in the future, but this seems fine for now. I just wonder if we should use tags like `harmony-plugin-X.X.X` instead of just `vX.X.X`? I think that would be more clear.

Comment (Ian2012) [reply .github/workflows/pypi-publish.yaml:11]: Sure, now that versions will be available on PyPI, we can name it as we want. Let me know if I can trigger the workflow on a PR event temporarily to make sure everything is working and to have a first version on PyPI

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Ian2012!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.



#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/openedx-k8s-harmony-maintainers`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

File: tutor-contrib-harmony-plugin/setup.py
Comment (Ian2012) [inline tutor-contrib-harmony-plugin/setup.py:29]: This is just to follow the standard on tutor plugins. Eventually, we would want to rename the folders but that's a large PR just to introduce this workflow. It's better to have a different PR for that

File: .github/workflows/pypi-publish.yaml
Comment (bradenmacdonald) [reply .github/workflows/pypi-publish.yaml]: Yes, go ahead and try it please :)

**Review by gabor-boros — APPROVED** [review]
Although I was never asked to review this, from my side it looks good!

**Comment by Ian2012** [comment]
Plugin is on pipy already, a PR for adding harmony to tpi is already in progress: https://github.com/overhangio/tpi/pull/5

I will close/open this PR as seems buggy

**Pull request closed without merging**
